src/bigquery/bigquery.py

```diff
@@ -0,0 +1,142 @@
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Dharmendra G Patel <dhpatel@redhat.com>
+#
+"""Bigquery implementation to read big data for manifest files."""
+import time
+from rudra import logger
+from rudra.data_store.aws import AmazonS3
+from src.config.settings import AWS_SETTINGS
+from src.bigquery.base import BigqueryBuilder, DataProcessing
+from src.bigquery.base_collector import BaseCollector
+from src.bigquery.maven_collector import MavenCollector
+from src.bigquery.npm_collector import NpmCollector
+from src.bigquery.pypi_collector import PypiCollector
+
+ECOSYSTEM_MANIFEST_MAP = {
+    'maven': 'pom.xml',
+    'npm': 'package.json',
+    'pypi': 'requirements.txt',
```

Comment:
How this would work incase if we want to use Poetry and Pipfile formats for python?

Reply:
We don't support this files yet, when we support them we need to modify this structure a bit to have multiple manifests for an ecosystem.

Reply:
You may need to change significantly incase if you plan to support multiple manifests per ecosystem.

Reply:
@arajkumar we can address the requirement as it comes, designing things for all possible future (uncertain) use-case will complicate current flow.